### PR TITLE
Zendesk mattlefevre/faraday upgrade

### PR DESCRIPTION
**Pull request by zendesk-mattlefevre**
## Description
As part of upgrading Samson to Ruby 3.0, we want to break up gem upgrades into smaller, more testable chunks. This PR is to bump the `faraday` gem, and all those that rely on it. 

### Risks
- Medium - we'll want to keep an eye on Github interactions for this one.

### CODE REVIEW

File: config/initializers/octokit.rb [hunk 1]
```diff
@@ -23,6 +23,8 @@ def on_complete(response)
   builder.response :logger, Rails.logger
   builder.use Octokit::Response::RaiseError
   builder.use Octokit::RedirectAsError
+  # Raise a RedirectLimitReached error if a request is redirected.
+  builder.use Octokit::Middleware::FollowRedirects, limit: 0
```
Comment (grosser) [inline config/initializers/octokit.rb:27]: seems to be doing the same as the above

File: test/support/git_repo_test_support.rb [hunk 2]
```diff
@@ -70,7 +70,8 @@ def create_submodule_repo
 
   def init_repo_commands
     <<-SHELL
-      git init
+      git init --initial-branch=master
```
Comment (grosser) [inline test/support/git_repo_test_support.rb:73]: nice!

File: test/models/outbound_webhook_test.rb [hunk 3]
```diff
@@ -91,21 +91,22 @@
...
+      refute_includes connection.builder.handlers, Faraday::Request::Authorization
     end
 
-    it "adds basic auth" do
+    it "adds auth with Basic authentication" do
       webhook.auth_type = "Basic"
       webhook.username = "adminuser"
       webhook.password = "abc123"
-      assert_equal connection.headers['Authorization'], 'Basic YWRtaW51c2VyOmFiYzEyMw=='
+      assert_includes connection.builder.handlers, Faraday::Request::Authorization
```
Comment (grosser) [inline test/models/outbound_webhook_test.rb:102]: I liked the old test more since it was very exact

File: Gemfile.lock [hunk 4]
```diff
@@ -771,4 +778,4 @@ RUBY VERSION
    ruby 2.7.6p219
 
 BUNDLED WITH
-   2.3.25
+   2.3.14
```
Comment (grosser) [inline Gemfile.lock:781]: the downgrade here is a bit sus but fine

**Review by grosser — APPROVED** [review]
looks about right!

**Comment by grosser** [comment]
nice, thx for the details

**Review by rsb — APPROVED** [review]

**Review by rsb — APPROVED** [review]

**Pull request merged**